CMakeLists.txt

```diff
@@ -110,6 +110,13 @@ else()
     message(STATUS "Building with IPO: off")
 endif()
 
+#
+# Build type flags
+#
+
+set(CMAKE_C_FLAGS_DEBUG "-O0 -g -ggdb")
```

Comment:
`-g3` gets you **a lot** better debugging information (at the cost of increased binary size) -- this allows debuggers to understand macros and similar.

I highly recommend adding `-fvar-tracking` and `-fvar-tracking-assignments` for better info even with optimization turned on.

Both of the above probably need  `-gdwarf-4` to really work properly.

Lastly, `-gsplit-dwarf` might be a good idea to speed up the link time on debug builds too and `-gz` would be good if the file sizes get very large.